src/Microsoft.AspNetCore.OData/Extensions/HttpRequestExtensions.cs

```diff
@@ -293,8 +293,11 @@ public static Uri GetNextPageLink(this HttpRequest request, int pageSize, object
                 uriBuilder.Port = request.Host.Port.Value;
             }
 
+            ODataOptions odataOptions = request.HttpContext.RequestServices.GetRequiredService<ODataOptions>();
+            ODataCompatibilityOptions compatibilityOptions = odataOptions.CompatibilityOptions;
+
```

Comment:
Consider a(n internal?) extension method on HttpRequest for getting compatibility options. Then shared code can call this method and we don't have to have different logic each place we get compatibility options.